### PR TITLE
Update privatelink.md

### PR DESCRIPTION
**Pull request by jubrad**
Updates documentation to note that privatelink is public preview https://github.com/MaterializeInc/cloud/issues/5190

### Motivation
  * This PR adds a known-desirable feature.
https://github.com/MaterializeInc/cloud/issues/5190

This is held up on a few testing tasks in the attached epic, just getting this ready. 

Also I believe we currently missing the private-preview note on this, so it might seem as if this is a reversion in feature lifecycle to those outside of materialize who are closely following our docs. 

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by morsapaes** [comment]
Thanks, @jubrad! It seems like we only mark it as private preview under [`CREATE CONNECTION`](https://materialize.com/docs/sql/create-connection/#aws-privatelink). I can push a commit that extends the annotation to all the required places.

**Review by morsapaes — APPROVED** [review]

File: doc/user/content/manage/invite-users.md
Comment (jubrad) [inline doc/user/content/manage/invite-users.md:1]: @morsapaes  this file was added in by bd82696f66810ce133f76a40c8ccabdd329859c , it seems unrelated, was it meant to be brought in here?

Comment (morsapaes) [reply doc/user/content/manage/invite-users.md:1]: `git add` using partial paths is as dangerous as you'd imagine. Thanks for catching this!
